Sources/Ink/Internal/CodeBlock.swift

```diff
@@ -13,20 +13,24 @@ internal struct CodeBlock: Fragment {
     private var code: String
 
     static func read(using reader: inout Reader) throws -> CodeBlock {
-        try require(reader.readCount(of: marker) == 3)
-
-        let language = reader.readUntilEndOfLine()
+        let startingMarkerCount = reader.readCount(of: marker)
+        try require(startingMarkerCount >= 3)
+        reader.discardWhitespaces()
+        var language = reader.readUntilEndOfLine()
+        while let l = language.last, l.isWhitespace {
+            language = language.dropLast()
+        }
         var code = ""
 
         while !reader.didReachEnd {
             if code.last == "\n", reader.currentCharacter == marker {
                 let markerCount = reader.readCount(of: marker)
 
-                if markerCount == 3 {
-                    code.removeLast()
+                if markerCount == startingMarkerCount {
                     break
                 } else {
                     code.append(String(repeating: marker, count: markerCount))
+                    if reader.didReachEnd { break } //maybe are at end of file? break for now?
```

Comment:
Curious, what kind of condition this is handling?

Reply:
the next line of code traps if the appended string was the last line of the markdown. So I added this double check to at least stop crashes. More rewriting of this code is required to handle many more CommonMark tests.